clients/cppsal/include/sal_node.h

```diff
@@ -0,0 +1,658 @@
+#pragma once
+
+#include "sal_data.h"
+#include "sal_exception.h"
+#include <cassert>
+
+#include "Poco/DateTimeFormatter.h"
+
+namespace sal
+{
+    /// tree-node structure supports std::string path like in file system
+    /// is there any third-party library provide this data structure?
+    namespace node
+    {
+        typedef enum
+        {
+            NODE_BRANCH,
+            NODE_LEAF
+        } NodeType;
+
+
+        // TODO: those global names should be removed
+        // static char JSON_CONTENT_REPORT[] = "report"; // should be named as "summary"

```

Comment:
Reports are not the same as a summary.
list() returns a report object that reports on the contents of a node.
get() returns the object contained in a node. A branch or a data object. The data object can be either a "full" object or a "summary" object as all data classes have two representations - the full data and a version containing only meta data (for use in tools visualising the data tree).

Reply:
I cant comment on cppsal. This pull request is for the csal interface.

Reply:
I think I got the difference this week, as I am doing unit test, using data from server.  For list() there is only one version of json, called report.

Reply:
You shouldn't need to infer how the system works from the server data, it's described in the datamodel and the SAL server rest API documentation.

Reply:
https://simple-access-layer.github.io/documentation/server.html describes all the returned data, broken down by request.